File: Frontend/src/services/Notificacion.service.js
import httpClient from "@/http-common";

// Crear una nueva notificación
const createNotificacion = (notificacion, token) => {
  return httpClient.post('/notificaciones', notificacion, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
};

// Obtener todas las notificaciones
const getAllNotificaciones = (token) => {
  return httpClient.get('/notificaciones', {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
};

// Actualizar una notificación por ID
const updateNotificacion = (id, notificacion, token) => {
  return httpClient.put(`/notificaciones/${id}`, notificacion, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
};

// Eliminar una notificación por ID
const deleteNotificacion = (id, token) => {
  return httpClient.delete(`/notificaciones/${id}`, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
};

export default {
  createNotificacion,
  getAllNotificaciones,
  updateNotificacion,
  deleteNotificacion,
};

File: Backend/demo/src/main/java/com/example/demo/Service/NotificacionService.java
package com.example.demo.Service;

import com.example.demo.Entity.Notificacion;
import com.example.demo.Repository.NotificacionRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.util.List;

@Service
public class NotificacionService {

    @Autowired
    private NotificacionRepository notificacionRepository;

    // Crear nueva notificación
    public Notificacion crearNotificacion(Notificacion notificacion) {
        return notificacionRepository.crear(notificacion);
    }

    // Obtener todas las notificaciones
    public List<Notificacion> obtenerNotificaciones(int page, int size) {
        return notificacionRepository.getAll(page, size);
    }

    // Actualizar notificación
    public String actualizarNotificacion(Notificacion notificacion, Integer id) {
        return notificacionRepository.update(notificacion, id);
    }

    // Eliminar notificación
    public void eliminarNotificacion(Integer id) {
        notificacionRepository.delete(id);
    }
}
